core/src/history.rs

```diff
@@ -99,4 +155,76 @@ mod tests {
             ],
         );
     }
+
+    #[test]
+    fn auction_settlement() {
+        let batch = BatchId(42);
+        let trades = vec![
+            Trade {
+                owner: Address::from_low_u64_be(1),
+                order_id: 0,
+                sell_token: 1,
+                buy_token: 2,
+                executed_sell_amount: 3,
+                executed_buy_amount: 4,
+            },
+            Trade {
+                owner: Address::from_low_u64_be(2),
+                order_id: 0,
+                sell_token: 2,
+                buy_token: 1,
+                executed_sell_amount: 4,
+                executed_buy_amount: 3,
+            },
+        ];
+        let solution = SolutionSubmission {
+            submitter: Address::from_low_u64_be(0),
+            utility: 1000.into(),
+            disregarded_utility: 100.into(),
+            burnt_fees: 1_000_000.into(),
+            last_auction_burnt_fees: 0.into(),
+            prices: vec![1, 2, 3],
+            token_ids_for_price: vec![0, 1, 2],
+        };
+
+        let history = {
+            let mut event_data = Vec::new();
+            event_data.extend(trades.iter().map(|trade| Event::Trade(trade.clone())));
+            event_data.push(Event::SolutionSubmission(solution.clone()));
+            event_data.push(token_listing(42));
+            event_data.extend(
+                trades
+                    .iter()
+                    .map(|trade| Event::TradeReversion(trade_as_reversion(trade))),
+            );
+            event_data.extend(trades.iter().map(|trade| Event::Trade(trade.clone())));
+            event_data.push(Event::SolutionSubmission(solution.clone()));
+
+            let mut events = EventRegistry::default();
+            for (i, event) in event_data.into_iter().enumerate() {
+                events.handle_event_data(
+                    EventData::Added(event),
+                    1337,
+                    i,
+                    block_hash(0),
+                    batch.next().as_timestamp(),
+                );
+            }
+
+            ExchangeHistory { events }
+        };
+
+        assert_eq!(
+            history.settlement_for_batch(batch).unwrap(),
+            Settlement { trades, solution },
+        );
+    }
+
+    #[test]
```

Comment:
Nice, we could test that trade reversion works as expected but I can sympathize if this is a lot of extra boilerplate code to add.

Reply:
I already test trade revision :smile: 
https://github.com/gnosis/dex-services/blob/bddbdfae9d2325709e9409b34e64dcdb70f97213/core/src/history.rs#L195-L199

I'll split it into two tests though, so its clearer.